core/src/test/java/org/jboss/set/mjolnir/archive/batch/MembershipRemovalBatchletTest.java

```diff
@@ -219,5 +224,55 @@ private void verifyBatchletProcessing(boolean removeFromOrg) throws Exception {
             WireMock.verify(0, deleteRequestedFor(urlEqualTo("/api/v3/orgs/testorg/members/TomasHofman")));
         }
     }
+    
+    @Test
+    public void testProcessRemoval_auditLog() throws Exception {
+        // configure that users should be unsubscribed from testorg
+        setUnsubscribeFromOrganization(true);
+        // configure GH API stubs
+        TestUtils.setupGitHubApiStubs();
+
+        // get a single removal to process
+        UserRemoval removal = em.createQuery("select r from UserRemoval r where ldapUsername = 'thofman'", UserRemoval.class)
+                .getSingleResult();
+        assertThat(removal.getStatus()).isNull();
+
+        // the #processRemoval() method doesn't open a transaction by itself, so need to do it here
+        em.getTransaction().begin();
+
+        // let the removal be processed
+        batchlet.processRemoval(removal);
+        em.getTransaction().commit();
+
+        // verify the processing ended successfully
+        em.refresh(removal);
+        assertThat(removal.getStatus()).isEqualTo(RemovalStatus.COMPLETED);
+
+        // verify audit log for unsubscribed orgs
+        List<UnsubscribedUserFromOrg> unsubscribedUsers =
+                em.createQuery("select u from UnsubscribedUserFromOrg u", UnsubscribedUserFromOrg.class)
+                        .getResultList();
+        assertThat(unsubscribedUsers).extracting("userRemoval.id", "githubUsername", "githubOrgName", "status")
+                .containsOnly(Tuple.tuple(removal.getId(), "TomasHofman", "testorg", UnsubscribeStatus.COMPLETED));
+
+        setUnsubscribeFromOrganization(false);
+
+        em.getTransaction().begin();
+
+        // let the removal be processed
+        batchlet.processRemoval(removal);
+        em.getTransaction().commit();
+
+        // verify the processing ended successfully
+        em.refresh(removal);
+        assertThat(removal.getStatus()).isEqualTo(RemovalStatus.COMPLETED);
```

Comment:
I think this part (lines 258 to 268) can be left out.  Both `UnsubscribedUserFromOrg` and `UnsubscribedUserFromTeam` records should have been created during the first call of `batchlet.processRemoval(removal)`, so no need to run it again.

The assertions bellow are correct. Just leave those.